### PR TITLE
[client] Client cannot be closed when reconnecting bugfix

### PR DESCRIPTION
**Pull request by mmatczuk**
To reproduce the bug:

1. start client with 
  * not existing server address
  * exponential back off strategy (infinite or just long enough)
1. stop client
1. client state is ClientConnecting, and client still tries to reconnect even though it was closed

### CODE REVIEW

**Comment by rjeczalik** [comment]
@mmatczuk Thanks!

File: client.go [hunk 1]
```diff
@@ -256,6 +256,16 @@ func (c *Client) Start() {
 			}
 
 			time.Sleep(dur)
+
+			// exit if closed
+			c.mu.Lock()
```
Comment (cihangir) [inline client.go]: we could change the type of `mu` to `RWMutex` instead of `Mutex`and use RLock and RUnlock here

Comment (mmatczuk) [inline client.go]: @cihangir I can do that next week no problem

Comment (rjeczalik) [inline client.go]: + while you at it please also create `closed() bol` / `setClosed(b bool)` and hide mutexes there

Comment (mmatczuk) [inline client.go]: :+1: sure

**Pull request merged**